org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/folding/LSPFoldingReconcilingStrategy.java

```diff
@@ -151,7 +150,7 @@ private void applyFolding(List<FoldingRange> ranges) {
 			if (ranges != null) {
 				Collections.sort(ranges, Comparator.comparing(FoldingRange::getEndLine));
 				for (FoldingRange foldingRange : ranges) {
-					updateAnnotation(modifications, deletions, existing, additions, foldingRange.getStartLine(),
+					updateAnnotation(deletions, existing, additions, foldingRange.getStartLine(),
```

Comment:
> we still need to call it, otherwise we would be breaking the customized code

Hm, I am not sure I am understand you here.  The modifications is always empty, because its not written anymore in `LSPFoldingReconcilingStrategy.updateAnnotations(existingAnnotation, newPos, deletions)`. The `updateAnnotation` which is called here in line 153 is a private method.

Theres no need to call the deprecated  `LSPFoldingReconcilingStrategy.updateAnnotations(existingAnnotation, newPos, modifications, deletions)` method here.

Reply:
> Hm, I am not sure I am understand you here. The modifications is always empty, because its not written anymore in `LSPFoldingReconcilingStrategy.updateAnnotations(existingAnnotation, newPos, deletions)`

Yes, the method as we have rewritten it will return empty annotations, but if some subclass subclassed the method, we still need to call it, otherwise we would break the customization.